### ISSUE TITLE
Self-reference not updated on refactor/rename

### ISSUE THREAD

**Issue by rasmuse** [dendronhq/dendron#421]
Thanks for addressing #406 so swiftly!

All the error messages are now gone, but I still have a couple of unexpected behaviors, for which I am now opening two separate issues. This is the first.

## My system

- Ubuntu 20.04.1 LTS
- VS Code v1.52
- Dendron 0.21.0

## How to reproduce

Create a note named `some-note` (using lookup -> create new, if that matters). Enter the following content (below the frontmatter):

```markdown
[[some-note]]
```

Save (and optionally close) the file.

Rename it using refactor or rename, e.g., to `some-note-renamed`.

## Expected result

A note `some-note-renamed` with the following content:

```markdown
[[some-note-renamed]]
```

## Actual result

The note is renamed to `some-note-renamed`, but its content is still

```markdown
[[some-note]]
```

**Comment by kpathakota** [comment]
Hmm... I'm struggling to see the value of self-referenced links. Is this for publishing 🧐? Nonetheless ... it is a valid bug that still repros.

**Comment by rasmuse** [comment]
Thanks @kpathakota for working on this.

No, I don't have an important use case for self-references. The reason I brought it up was that for me it is very important that rename/refactor is dependable and well defined. I did have some issues previously with rename/refactor (see #406) and then I tried to break down very carefully what behavior I expect from it. This issue as well as #422 were the remaining open questions after the error messages in #406 were resolved.

I leave it up to someone else to decide whether this issue is actually worth solving. But _in principle_ I insist that self-references should be handled as suggested above.
